fiftyone/utils/cvat.py

```diff
@@ -2377,6 +2425,1904 @@ def write(
+                    d["reason"],
+                    d["_content"],
+                )
+            )
+
+    def delete(self, url):
+        response = self._session.delete(url, verify=False)
+        self._raise_response_errors(response)
+        return response
+
+    def get(self, url):
+        response = self._session.get(url, verify=False)
+        self._raise_response_errors(response)
+        return response
+
+    def patch(self, url, auth=None, data=None, files=None, json=None):
+        kwargs = {"url": url, "verify": False}
+        if auth is not None:
+            kwargs["auth"] = auth
+        if data is not None:
+            kwargs["data"] = data
+        if files is not None:
+            kwargs["files"] = files
+        if json is not None:
+            kwargs["json"] = json
+
+        response = self._session.patch(**kwargs)
+        self._raise_response_errors(response)
+        return response
+
+    def post(self, url, auth=None, data=None, files=None, json=None):
+        kwargs = {"url": url, "verify": False}
+        if auth is not None:
+            kwargs["auth"] = auth
+        if data is not None:
+            kwargs["data"] = data
+        if files is not None:
+            kwargs["files"] = files
+        if json is not None:
+            kwargs["json"] = json
+
+        response = self._session.post(**kwargs)
+        self._raise_response_errors(response)
+        return response
+
+    def put(self, url, auth=None, data=None, files=None, json=None):
+        kwargs = {"url": url, "verify": False}
+        if auth is not None:
+            kwargs["auth"] = auth
+        if data is not None:
+            kwargs["data"] = data
+        if files is not None:
+            kwargs["files"] = files
+        if json is not None:
+            kwargs["json"] = json
+
+        response = self._session.put(**kwargs)
+        self._raise_response_errors(response)
+        return response
+
+    def setup(self):
+        """Performs any necessary setup for the API."""
+        if self._auth is None:
+            self._auth = self.get_username_password("CVAT")
+        self._session = requests.Session()
+        response = self.post(self.login_url, data=self._auth)
+        if "csrftoken" in response.cookies:
+            self._session.headers["X-CSRFToken"] = response.cookies[
+                "csrftoken"
+            ]
+
+    def get_username_password(self, host=""):
+        username = fo.annotation_config.cvat_username
+        password = fo.annotation_config.cvat_password
+
+        if username is None or password is None:
+            logger.info(
+                "No config or environment variables found for "
+                "authentication. Please enter CVAT login information. Set the "
+                "environment variables `FIFTYONE_CVAT_USERNAME` and "
+                "`FIFTYONE_CVAT_PASSWORD` to avoid this in the future."
+            )
+            self.prompt_username_password(host="CVAT")
```

Comment:
I think the you want to return here with the prompt `dict`?

Reply:
Ah, yup!